### PR TITLE
Fixes and Adds Item on Npc

### PR DESCRIPTION
**Pull request by Lmctruck30**
- Fixes the packet size issue
- Adds Item on Npc message and decoder
- #304 Issue

### CODE REVIEW

File: game/src/main/org/apollo/game/message/impl/ItemOnNpcMessage.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+/**
+ * A {@link Message} sent by the client when a player uses an item on a npc.
+ * 
+ * @author Lmctruck30
+ */
+
+public final class ItemOnNpcMessage extends Message {
+
+	private int itemId;
```
Comment (garyttierney) [inline game/src/main/org/apollo/game/message/impl/ItemOnNpcMessage.java]: Make these fields final.

**Comment by garyttierney** [comment]
LGTM besides the 2 comments.

**Comment by garyttierney** [comment]
Actually, also, which packet sizes does your first commit fix?

**Comment by Lmctruck30** [comment]
@garyttierney  57

**Comment by Lmctruck30** [comment]
@garyttierney @ryleykimmel 
Added another quick fix to the source. Can this be accepted yet?

**Comment by garyttierney** [comment]
@Lmctruck30 I made one more comment on the ReportAbuseMessage class https://github.com/apollo-rsps/apollo/pull/306/files/d95e8931f62bdeca5c889fc99fc2c4e0788a1a12#diff-ad489f93dbc2cdf2490e0dc79f686ab1

**Comment by ryleykimmel** [comment]
Also if you don't mind can you make your documentation more consistent?

**Comment by Lmctruck30** [comment]
@garyttierney @ryleykimmel All changes have been made and those fields have already been turned into final in the previous commit.

**Comment by garyttierney** [comment]
@Lmctruck30 @ryleykimmel Few nitpicks, but nothing I can't fix before squashing and merging.  LGTM.  One thing I want to check before merging this is this: are rule ID's very much just static identifiers, or do we need to map them to something with better semantics?

Should a plugin listen on events for "rule_id == 5", and have its own definitions for rules, or should a plugin listen on events for "rule_id == ReportAbuseRule.INAPPROPRIATE_MESSAGE" ?

**Comment by Lmctruck30** [comment]
@garyttierney best information on this would be to log the report with chat history, pm history, trade/duel/attack/action history.

File: game/src/main/org/apollo/game/message/impl/ItemOnNpcMessage.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
+package org.apollo.game.message.impl;
+
+import org.apollo.net.message.Message;
+
+/**
+ * A {@link Message} sent by the client when a player uses an item on a npc.
+ *
+ * @author Lmctruck30
+ */
+
```
Comment (ryleykimmel) [inline game/src/main/org/apollo/game/message/impl/ItemOnNpcMessage.java]: Remove the space here, please

File: game/src/main/org/apollo/game/message/impl/ReportAbuseMessage.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
+package org.apollo.game.message.impl;
+
+import org.apollo.net.message.Message;
+
+/**
+ * A {@link Message} sent by the client when a player reports another player.
+ *
+ * @author Lmctruck30
+ */
+
```
Comment (ryleykimmel) [inline game/src/main/org/apollo/game/message/impl/ReportAbuseMessage.java]: Same

File: game/src/main/org/apollo/game/release/r317/Release317.java [hunk 4]
```diff
@@ -61,12 +61,13 @@
...
 	 */
-	public static final int[] PACKET_LENGTHS = { 0, 0, 0, 1, -1, 0, 0, 0, 0, 0, // 0
+	public static final int[] PACKET_LENGTHS = {
+			0, 0, 0, 1, -1, 0, 0, 0, 0, 0, // 0
 			0, 0, 0, 0, 8, 0, 6, 2, 2, 0, // 10
 			0, 2, 0, 6, 0, 12, 0, 0, 0, 0, // 20
 			0, 0, 0, 0, 0, 8, 4, 0, 0, 2, // 30
 			2, 6, 0, 6, 0, -1, 0, 0, 0, 0, // 40
-			0, 0, 0, 12, 0, 0, 0, 0, 8, 0, // 50
+			0, 0, 0, 12, 0, 0, 0, 8, 8, 0, // 50
```
Comment (ryleykimmel) [inline game/src/main/org/apollo/game/release/r317/Release317.java]: Can you show the client code that makes this change required? (Preferably a deob)

File: game/src/main/org/apollo/game/release/r317/ReportAbuseMessageDecoder.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
+package org.apollo.game.release.r317;
+
+import org.apollo.game.message.impl.ReportAbuseMessage;
+import org.apollo.net.codec.game.DataType;
+import org.apollo.net.codec.game.GamePacket;
+import org.apollo.net.codec.game.GamePacketReader;
+import org.apollo.net.release.MessageDecoder;
+
+public final class ReportAbuseMessageDecoder extends MessageDecoder<ReportAbuseMessage> {
```
Comment (ryleykimmel) [inline game/src/main/org/apollo/game/release/r317/ReportAbuseMessageDecoder.java]: Documentation, please

**Comment by Major-** [comment]
Dealt with the comments, thanks for the original PR!

**Pull request closed without merging**